lib/runtime/offchain/httpset.go

```diff
@@ -48,10 +52,35 @@ func (b requestIDBuffer) put(i int16) error {
 	}
 }
 
+// Request holds the request object and update the invalid and waiting status whenever
+// the request starts or is waiting to be read
+type Request struct {
```

Comment:
I don't believe we can get improvements storing these values on `http.Request's context` the reason we had these values is to avoid operations over already ongoing or finalized HTTP calls

Reply:
It's not for improvements except to use `*http.Request` directly. Most http routers store stuff in the http.Request's context, I think it would fit nicely, so we would have one less type to manage ourselves.

Example

```go
type contextKey string

const waitingKey contextKey = "waiting"

// ...

// to set it
request := http.NewRequest(...)
ctx := context.WithValue(request.Context(), waitingKey, false)
request = request.WithContext(ctx)

// to retrieve it
waiting := request.Context().Value(ContextUserKey).(bool)
```

Reply:
@qdm12, done!